### PR TITLE
Support recursive extended tsconfigs

### PR DESCRIPTION
**Pull request by alan-agius4**

### CODE REVIEW

**Comment by Toxicable** [comment]
I'm 70% sure this logic already exists withing `ts`'s apis, I dont see a need to reimplement it.
The Angular CLI guys have done this already.
I **think** this is the function which does it here.
https://github.com/angular/angular-cli/blob/master/packages/angular_devkit/build_angular/src/angular-cli-files/utilities/read-tsconfig.ts#L14

**Comment by alan-agius4** [comment]
@Toxicable , the thing is that TS API, only support extending for `TypeScript` options, `compilerOptions` anything else such as `bazelOptions` and `angularCompilerIOptions` will not be merged.

File: internal/tsc_wrapped/tsconfig.ts [hunk 1]
```diff
@@ -258,49 +258,57 @@ export function parseTsconfig(
...
+        return {error};
+      }
+
+      // Allow Bazel users to control some of the bazel options.
+      // Since TypeScript's "extends" mechanism applies only to "compilerOptions"
+      // we have to repeat some of their logic to get the user's bazelOptions.
+      const baseConfig = existingConfig || config;
+    
+      if (existingConfig) {
+        baseConfig.bazelOptions = {
```
[the remark this answers is not in the document]
Comment (alan-agius4) [reply internal/tsc_wrapped/tsconfig.ts]: Yes, done

File: internal/tsc_wrapped/tsconfig.ts [hunk 2]
```diff
@@ -258,49 +258,57 @@ export function parseTsconfig(
...
+          ...config.bazelOptions,
+          ...baseConfig.bazelOptions
```
[the remark this answers is not in the document]
Comment (alan-agius4) [reply internal/tsc_wrapped/tsconfig.ts]: addressed.

**Comment by alexeagle** [comment]
mailed in cl/235367520 to @kyliau

**Pull request closed without merging**